syft/pkg/cataloger/golang/parse_go_bin.go

```diff
@@ -1,26 +1,53 @@
+//nolint
 package golang
 
 import (
-	"bufio"
+	"bytes"
+	"debug/elf"
+	"debug/macho"
+	"debug/pe"
+	"errors"
 	"fmt"
 	"io"
+	"runtime/debug"
 	"strings"
 
+	"github.com/anchore/syft/internal/log"
 	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/golang/internal/xcoff"
 	"github.com/anchore/syft/syft/source"
 )
 
-const (
-	packageIdentifier = "dep"
-	replaceIdentifier = "=>"
+const GOARCH = "GOARCH"
+
+var (
+	// errUnrecognizedFormat is returned when a given executable file doesn't
+	// appear to be in a known format, or it breaks the rules of that format,
+	// or when there are I/O errors reading the file.
+	errUnrecognizedFormat = errors.New("unrecognized file format")
 )
 
-type exeOpener func(file io.ReadCloser) ([]exe, error)
+func makeGoMainPackage(mod *debug.BuildInfo, arch string, location source.Location) pkg.Package {
+	gbs := getBuildSettings(mod.Settings)
+	main := newGoBinaryPackage(&mod.Main, mod.GoVersion, arch, location, gbs)
+	main.Version = ""
+
+	if v, ok := gbs["vcs.revision"]; ok {
+		main.Version = v
```

Comment:
_not a comment to change anything on this PR, just a thought_: we could toy with changing the version to be in the format of `v0.0.0-<date>-<commit>` if we could derive the date correctly from the build info (which I'm not certain is possible). This would make it consistent to go module version formatting.